WebClient/Scripts/temperatureControlView.js

```diff
@@ -0,0 +1,503 @@
+﻿
+L.Control.Temp = L.Control.extend({
+    options: {
+        collapsed: false,
+        position: 'topleft',
+        autoZIndex: true,
+        height: 150,
+        width: 150,
+    },
+
+    /// leaflet constructor
+    initialize: function (options) {
+        L.setOptions(this, options);
+        this.parentContainer = options.element;
+    },
+
+    onAdd: function (map) {
+        this.active = true;
+        this._initLayout();
+        this._map = map;
+        return this._container;
+    },
+
+    onRemove: function () {
+        this.active = false;
+    },
+
+    _initLayout: function () {
+        this._container = this.parentContainer;//L.DomUtil.create('div', 'd3-control-temperatureControl');
+
+        //Code for making the div draggable
+        // add drag support to main div
+        //this._draggable = new L.Draggable(this._container);
+        //this._draggable.ref = this;
+
+        //this._draggable.enable();
+
+        this._temperatureDiv = L.DomUtil.create('div', "temperatureDiv");
+        this._container.appendChild(this._temperatureDiv);
+
+        //disabling propagation
+        L.DomEvent.disableClickPropagation(this._container);
+        L.DomEvent.disableScrollPropagation(this._container);
+
+        // use context menu event to return to "live" state
+        this._container.addEventListener('contextmenu', this._handleContextMenu);
+
+        //--------------------Variable declarations------------------------
+        //svg image vars
+        var width = 90;
+        var height = 160;
+        this.borderWidth = 1;
+
+        //general x-coordinate var
+        this.xPosForBaseComponents = (width / 2.5);
+
+        //capillary tube vars
+        this.tubeWidth = 12;
+        this.tubeRectY = (height / 8);
+        this.tubeHeight = ((3.1 * height) / 4) - this.tubeRectY;
+        this.tubeRectX = this.xPosForBaseComponents - (this.tubeWidth / 2);
+
+        this.tubeBulbPixelOverlap = 3; //the spherical bulb overlaps the capillary tube by defined pixels
+
+        //mercury vals
+        this.mercuryColor_red = "#E50000";
```

Comment:
It is better to use CSS for such kind of stylings.

Reply:
Moved this to the CSS for changing the color of the mercury bar but still these variables will exist in the code because mercury color gradient for the mercury bulb could not be set in the CSS file.